### DISCUSSION TITLE
ControlNet Speed Performance Collection

### DISCUSSION THREAD

**Discussion by lllyasviel** [Mikubill/sd-webui-controlnet#2061] [General]
Because SDXL is a relatively large model, we need to make sure that the current codes are correctly using your GPU.

We invite you to share some screenshots like this from your webui here:

![image](https://github.com/Mikubill/sd-webui-controlnet/assets/19834515/2a3ec866-640e-404b-a332-cd6a0d6e823c)

The “time taken” will show how much time you spend on generating an image. The “sys” will show the VRAM of your GPU.

We highly appreciate your help if you can share a screenshot in this format:

    GPU (like RGX 4096, RTX 3080, ...)
    -cmdflag (like --medvram-sdxl, --lowvram, ...)
    Then put an image

(To avoid the influence of downloading preprocessors, we recommend to use canny for test. If possible, you may use 1024 resolution and sdxl models. Feel free to use any canny control model as you wish.)

I will start with my own test as an example. Looking forward to your test screenshots.

**Comment by lllyasviel** [comment]
RTX 3070ti laptop GPU
--lowvram --xformers
![image](https://github.com/Mikubill/sd-webui-controlnet/assets/19834515/92958e5c-30ce-4d38-9249-c8f0ba046d94)

**Comment by lllyasviel** [comment]
RTX 3070ti laptop GPU
--medvram-sdxl --xformers
![image](https://github.com/Mikubill/sd-webui-controlnet/assets/19834515/05d4d7d0-53d3-4d1d-bf08-7fa2ca35dfd4)

**Comment by lllyasviel** [comment]
RTX 3070ti laptop GPU
--medvram-sdxl --xformers
![image](https://github.com/Mikubill/sd-webui-controlnet/assets/19834515/3f7582f4-c637-4bc8-921a-c45e1e4e4489)

**Comment by lllyasviel** [comment]
RTX 3070ti laptop GPU
--medvram-sdxl --xformers
![image](https://github.com/Mikubill/sd-webui-controlnet/assets/19834515/9c01d5bf-a22a-444d-82fd-3f68295b517b)

**Comment by lllyasviel** [comment]
RTX 4090
--xformers
![image](https://github.com/Mikubill/sd-webui-controlnet/assets/19834515/16955a08-e3c9-4f22-b392-1c13f851760d)

**Comment by lllyasviel** [comment]
3060 laptop GPU
--lowvram --xformers
![image](https://github.com/Mikubill/sd-webui-controlnet/assets/19834515/d142bc08-5fea-400f-a203-4aaf71b571af)

**Comment by youyegit** [comment]
RTX 3080 12G 
--xformers --medvram-sdxl
![image](https://github.com/Mikubill/sd-webui-controlnet/assets/131540443/930de872-6534-449f-8cae-2853ddc63db7)
![image](https://github.com/Mikubill/sd-webui-controlnet/assets/131540443/0a61f07e-ab69-4ee5-aa62-064e264db491)

**Comment by w-e-w** [comment]
RTX 3090 24G --xformer
![image](https://github.com/Mikubill/sd-webui-controlnet/assets/40751091/d01a9111-943a-4c67-bcd6-af6babac1a3a)

**Comment by manticorp** [comment]
RTX 4090 24GB

```--opt-sdp-no-mem-attention --autolaunch --gradio-img2img-tool color-sketch```

![image](https://github.com/Mikubill/sd-webui-controlnet/assets/3750419/82ce7c4d-3821-4b94-895c-0bc6357425d9)
![image](https://github.com/Mikubill/sd-webui-controlnet/assets/3750419/72d7068a-cd5a-4b4c-8fd6-167f8ea08320)

![image](https://github.com/Mikubill/sd-webui-controlnet/assets/3750419/ae946b9f-e4e5-4ba6-88cc-457eb3b41964)

**Comment by manticorp** [comment]
RTX 4090 24GB

```--opt-sdp-no-mem-attention --autolaunch --gradio-img2img-tool color-sketch```

![image](https://github.com/Mikubill/sd-webui-controlnet/assets/3750419/cfd37a40-f16c-42ca-a506-cbe32474e250)

**Comment by manticorp** [comment]
RTX 4090 24GB

```--opt-sdp-no-mem-attention --autolaunch --gradio-img2img-tool color-sketch```

![image](https://github.com/Mikubill/sd-webui-controlnet/assets/3750419/74ac88a9-1745-429e-8403-0133f3f56047)

**Comment by HMRMike** [comment]
RTX 3090 24GB 
--xformers
![image](https://github.com/Mikubill/sd-webui-controlnet/assets/13288923/10091884-544e-4e26-a12e-2c50b2bfba2d)

**Comment by lllyasviel** [comment]
Sep 6:
I looked at all above. All cases are within expected speed.
Please share more screenshots to help us investigating these cases

**Comment by ClashSAN** [comment]
RTX 3050 laptop GPU
`--opt-sdp-no-mem-attention --lowvram --theme dark`

![e (1)](https://github.com/Mikubill/sd-webui-controlnet/assets/98228077/12178415-85b5-4c99-a7ed-4ce8d5da45a2)
![e (10)](https://github.com/Mikubill/sd-webui-controlnet/assets/98228077/82235ecf-52b8-4bb8-97c0-aadbd32f84f4)
![e (11)](https://github.com/Mikubill/sd-webui-controlnet/assets/98228077/41696027-1992-48aa-89c6-aa0a9d6d278c)
![e (13)](https://github.com/Mikubill/sd-webui-controlnet/assets/98228077/bf41a0f4-a0a0-4d07-85d3-4117c61c5d63)
![g (2)](https://github.com/Mikubill/sd-webui-controlnet/assets/98228077/fe7702be-eec3-4d01-8f32-44f3e86c791e)
![g (3)](https://github.com/Mikubill/sd-webui-controlnet/assets/98228077/5bfbfca0-e0e4-4d95-bb4b-b1121deff44f)
![g](https://github.com/Mikubill/sd-webui-controlnet/assets/98228077/78d54774-af1e-42af-8305-020f5802038f)

**Comment by JetsprintRacer** [comment]
RTX 3080 Ti
For some reason this thing is slow as snail when I use SDXL Reference, and "OutOfMemoryError: CUDA out of memory" when I try to use Sargezt XL Softedge. Estimated time is 30-60 minutes sometimes. How in the world you managed to render pictures in 36 seconds without any problems while using a Reference model - that's quite a mistery for me. Also, a lot of "Out of memory errors", when I'm interrupting generations too often - probably, allocated memory unload problem. Revision, IP-Adapter and Sai Canny seem to work fine. I guess... Also, first time generation with new preprocessor image in SDXL is quite long.
![Screenshot 2023-09-07 205613](https://github.com/Mikubill/sd-webui-controlnet/assets/144376283/f0abfd24-3d10-4024-8a90-204a2343a184)
![Screenshot 2023-09-07 213359](https://github.com/Mikubill/sd-webui-controlnet/assets/144376283/46ca14de-b5b0-405f-822d-f43d0a1c74fb)
![Screenshot 2023-09-07 213706](https://github.com/Mikubill/sd-webui-controlnet/assets/144376283/595e81b4-651c-43eb-8dbb-069d81bd8a3a)
![Screenshot 2023-09-07 213902](https://github.com/Mikubill/sd-webui-controlnet/assets/144376283/cd47738a-518c-48bf-8b1c-81e5746d91d7)
![Screenshot 2023-09-07 214107](https://github.com/Mikubill/sd-webui-controlnet/assets/144376283/8216b21f-29f4-4e23-9f35-3eb88acdc448)
![Screenshot 2023-09-07 214700](https://github.com/Mikubill/sd-webui-controlnet/assets/144376283/4649c764-af61-4eb8-bd78-2c2f9602499f)

**Reply by lllyasviel** [reply]
`-cmdflag` is missing. This problem should be solved by adding --medvram-sdxl

**Reply by gumpngreen** [reply]
What's the difference between --medvram and --medvram-sdxl?

**Reply by JetsprintRacer** [reply]
First one affects all checkpoints, second one affects only SDXL checkpoints.

**Comment by boranali** [comment]
rtx 1050 4 GB laptop :D

cinematic still photography of a perfect modern sports car. best quality,high resolution,raw photo,designed by a master,flawless. emotional,harmonious,vignette,highly detailed,high budget,bokeh,cinemascope,moody,epic,gorgeous,film grain,grainy,

Negative prompt: drawing,painting,illustration. worst quality,low quality,low resolution. imperfect,bad design. anime,cartoon,graphic,rendered,text,crayon,graphite,abstract,glitch,deformed,mutated,ugly,disfigured,
Steps: 40, Sampler: DPM++ 2M SDE, CFG scale: 7, Seed: 3836892232, Size: 1024x1024, Model hash: 0f1b80cfe8, Model: dreamshaperXL10_alpha2Xl10, RNG: CPU, SGM noise multiplier: True, Version: v1.6.0

Time taken: 9 min. 11.9 sec.

A: 3.41 GB, R: 7.14 GB, Sys: 4.0/4 GB (100.0%)

**Comment by MoonRide303** [comment]
RTX 4080 (desktop), base + refiner, double Control-LoRA (rank 128, canny + depth).

COMMANDLINE_ARGS:
```
--xformers --medvram --no-half-vae
```

Prompt:
```
cinematic still perfect supercar . emotional, harmonious, vignette, highly detailed, high budget, bokeh, cinemascope, moody, epic, gorgeous, film grain, grainy
Negative prompt: anime, cartoon, graphic, text, painting, crayon, graphite, abstract, glitch, deformed, mutated, ugly, disfigured
Steps: 30, Sampler: DPM++ 2M SDE Karras, CFG scale: 7, Seed: 0, Size: 1248x832, Model hash: e6bb9ea85b, Model: sd_xl_base_1.0_0.9vae, Clip skip: 2, RNG: CPU, ControlNet 0: "Module: canny, Model: control-lora-canny-rank128 [c910cde9], Weight: 0.5, Resize Mode: Resize and Fill, Low Vram: False, Processor Res: 1024, Threshold A: 100, Threshold B: 200, Guidance Start: 0, Guidance End: 0.3, Pixel Perfect: False, Control Mode: Balanced", ControlNet 1: "Module: none, Model: control-lora-depth-rank128 [df51c1c8], Weight: 0.5, Resize Mode: Resize and Fill, Low Vram: False, Guidance Start: 0, Guidance End: 0.3, Pixel Perfect: False, Control Mode: Balanced", SGM noise multiplier: True, Refiner: sd_xl_refiner_1.0_0.9vae [8d0ce6c016], Refiner switch at: 0.67, Version: v1.6.0
```

Results, 1st run:
```
Time taken: 36.8 sec.

A: 6.92 GB, R: 12.52 GB, Sys: 15.0/15.9922 GB (93.8%)
```

Results, 2nd run:
```
Time taken: 27.7 sec.

A: 6.95 GB, R: 12.55 GB, Sys: 15.1/15.9922 GB (94.1%)
```

Input (same to both CNs):
![image](https://github.com/Mikubill/sd-webui-controlnet/assets/130458190/6dae8a58-44d4-4d61-8e6b-716b23241c49)

Output:
![image](https://github.com/Mikubill/sd-webui-controlnet/assets/130458190/e5748198-912e-4b59-a88e-1a4b75e39e3f)

**Comment by kasimgu2000** [comment]
Not sure whats going on here....any ideas?

RTX 3080 Ti 12GB

--medvram-sdxl --xformers --no-half --autolaunch

![memory_overload](https://github.com/Mikubill/sd-webui-controlnet/assets/145486428/6ad6fca2-b0d6-4e27-b5e9-fb2bf49df836)

**Reply by VeraLapsa** [reply]
I'm on a 3070-8GB and if I run the latest nvidia drivers (I think 536.99) my generation time skyrockets because of the memory management it does (offloads to system memory). I don't know if it'll help but downgrading to the 531.61 drivers fixed long generation times for me.

**Reply by MoonRide303** [reply]
536.67 work fine for me.

**Reply by kasimgu2000** [reply]
no luck with changing drivers....might try a complete re-install of everything...

**Reply by MoonRide303** [reply]
@kasimgu2000 You can also try with different COMMANDLINE_ARGS in webui-user.bat, these work reasonably for me: `set COMMANDLINE_ARGS=--xformers --medvram --no-half-vae`.

**Reply by kasimgu2000** [reply]
MoonRide303 - Yes! This particular combination of parameters seem to have worked! Thank you =)
Now for the next challenge: trying to get the Lineart and Reference Controlnets actually do something...
